src/git-repository-async.js

```diff
@@ -0,0 +1,871 @@
+  //  * `remotes` An {Array} of remote reference names.
+  //  * `tags`    An {Array} of tag reference names.
+  getReferences (_path) {
+    return this.getRepo(_path)
+      .then(repo => repo.getReferences(Git.Reference.TYPE.LISTALL))
+      .then(refs => {
+        const heads = []
+        const remotes = []
+        const tags = []
+        for (const ref of refs) {
+          if (ref.isTag()) {
+            tags.push(ref.name())
+          } else if (ref.isRemote()) {
+            remotes.push(ref.name())
+          } else if (ref.isBranch()) {
+            heads.push(ref.name())
+          }
+        }
+        return {heads, remotes, tags}
+      })
+  }
+
+  // Public: Get the SHA for the given reference.
+  //
+  // * `reference` The {String} reference to get the target of.
+  // * `path` An optional {String} path in the repo to get the reference target
+  //   for. Only needed if the repository contains submodules.
+  //
+  // Returns a {Promise} which resolves to the current {String} SHA for the
+  // given reference.
+  getReferenceTarget (reference, _path) {
+    return this.getRepo(_path)
+      .then(repo => Git.Reference.nameToId(repo, reference))
+      .then(oid => oid.tostrS())
+  }
+
+  // Reading Status
+  // ==============
+
+  // Public: Resolves true if the given path is modified.
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is modified.
+  isPathModified (_path) {
+    return this._filterStatusesByPath(_path).then(statuses => {
+      return statuses.filter(status => status.isModified()).length > 0
+    })
+  }
+
+  // Public: Resolves true if the given path is new.
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is new.
+  isPathNew (_path) {
+    return this._filterStatusesByPath(_path).then(statuses => {
+      return statuses.filter(status => status.isNew()).length > 0
+    })
+  }
+
+  // Public: Is the given path ignored?
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is ignored.
+  isPathIgnored (_path) {
+    return this.repoPromise
+      .then(repo => Git.Ignore.pathIsIgnored(repo, _path))
+      .then(ignored => Boolean(ignored))
+  }
+
+  // Get the status of a directory in the repository's working directory.
+  //
+  // * `directoryPath` The {String} path to check.
+  //
+  // Returns a {Promise} resolving to a {Number} representing the status. This
+  // value can be passed to {::isStatusModified} or {::isStatusNew} to get more
+  // information.
+  getDirectoryStatus (directoryPath) {
+    let relativePath
+    // XXX _filterSBD already gets repoPromise
+    return this.repoPromise
+      .then(repo => {
+        relativePath = this.relativize(directoryPath, repo.workdir())
```

Comment:
This variable appears to be only used in this promise but is declared outside this scope on line 402, could the `let` move here instead?